deepchem/feat/vocabulary_builders/grover_vocab.py

```diff
@@ -0,0 +1,414 @@
+        self.specials = ('<pad>', '<other>')
+        self.min_freq = 1
+        self.max_size = None
+        self.itos = list(self.specials)
+        self.stoi = self._make_reverse_mapping(self.itos)
+        self.pad_index = 0
+        self.other_index = 1
+
+    def build(self, dataset: Dataset) -> None:
+        """Builds vocabulary
+
+        Parameters
+        ----------
+        dataset: dc.data.Dataset
+            A dataset object with SMILEs strings in X attribute.
+        """
+        counter: Dict[str, int] = Counter()
+        for x, _, _, _ in dataset.itersamples():
+            smiles = x[0]
+            mol = Chem.MolFromSmiles(smiles)
+            for bond in mol.GetBonds():
+                v = self.bond_to_vocab(mol, bond)
+                counter[v] += 1
+
+        # sort first by frequency, then alphabetically
+        words_and_frequencies = sorted(counter.items(), key=lambda tup: tup[0])
+        words_and_frequencies.sort(key=lambda tup: tup[1], reverse=True)
+        for word, freq in words_and_frequencies:
+            self.itos.append(word)
+        self.stoi = self._make_reverse_mapping(self.itos)
+
+    def save(self, fname: str) -> None:
+        """Saves a vocabulary in json format
+
+        Parameter
+        ---------
+        fname: str
+            Filename to save vocabulary
+        """
+        vocab = {'stoi': self.stoi, 'itos': self.itos}
+        with open(fname, 'w') as f:
+            json.dump(vocab, f)
+
+    @classmethod
+    def load(cls, fname: str) -> 'GroverBondVocabularyBuilder':
+        """Loads vocabulary from the specified json file
+
+        Parameters
+        ----------
+        fname: str
+            JSON file containing vocabulary
+
+        Returns
+        -------
+        vocab: GroverAtomVocabularyBuilder
+            A grover atom vocabulary builder which can be used for encoding
+        """
+        with open(fname, 'r') as f:
+            data = json.load(f)
+        vocab = cls()
+        vocab.stoi, vocab.itos = data['stoi'], data['itos']
+        return vocab
+
+    @staticmethod
+    def bond_to_vocab(mol: RDKitMol, bond: RDKitBond):
+        """Convert bond to vocabulary.
+
+        The algorithm considers only one-hop neighbor atoms.
+
+        Parameters
+        ----------
+        mol: Molecule
+            the molecular.
+        atom: rdchem.Atom
+            the target atom.
+
+        Returns
+        -------
+        vocab: str
+            the generated bond vocabulary with its contexts.
+
+        Example
+        -------
+        >>> from rdkit import Chem
+        >>> mol = Chem.MolFromSmiles('[C@@H](C)C(=O)O')
+        >>> GroverBondVocabularyBuilder.bond_to_vocab(mol, mol.GetBondWithIdx(0))
+        '(SINGLE-STEREONONE-NONE)_C-(SINGLE-STEREONONE-NONE)1'
+        >>> GroverBondVocabularyBuilder.bond_to_vocab(mol, mol.GetBondWithIdx(2))
+        '(DOUBLE-STEREONONE-NONE)_C-(SINGLE-STEREONONE-NONE)2'
+        """
+        nei: Dict[str, int] = Counter()
```

Comment:
What does `nei` mean? Better in general to use descriptive variable names since hard to read otherwise

Reply:
updated to descriptive name